python/grass/temporal/univar_statistics.py

```diff
@@ -109,19 +124,34 @@ def print_gridded_dataset_univar_statistics(
                     _("Unable to get statistics for 3d raster map" " <%s>") % id
                 )
             continue
-
-        string += str(id) + fs + str(start) + fs + str(end)
-        string += fs + str(stats["mean"]) + fs + str(stats["min"])
-        string += fs + str(stats["max"]) + fs + str(stats["mean_of_abs"])
-        string += fs + str(stats["stddev"]) + fs + str(stats["variance"])
-        string += fs + str(stats["coeff_var"]) + fs + str(stats["sum"])
-        string += fs + str(stats["null_cells"]) + fs + str(stats["cells"])
-        string += fs + str(int(stats["cells"]) - int(stats["null_cells"]))
-        if extended is True:
-            string += fs + str(stats["first_quartile"]) + fs + str(stats["median"])
-            string += (
-                fs + str(stats["third_quartile"]) + fs + str(stats["percentile_90"])
-            )
+        eol = ""
+
+        for idx, stats_kv in enumerate(univar_stats.split(";")):
+            stats = gscript.utils.parse_key_val(stats_kv)
+            string += str(id) + fs + str(start) + fs + str(end)
+            if zones:
+                if idx == 0:
+                    zone = str(stats["zone"])
+                    string = ""
+                    continue
+                string += fs + str(zone)
+                if "zone" in stats:
+                    zone = str(stats["zone"])
+                    eol = "\n"
+                else:
+                    eol = ""
+            string += fs + str(stats["mean"]) + fs + str(stats["min"])
+            string += fs + str(stats["max"]) + fs + str(stats["mean_of_abs"])
+            string += fs + str(stats["stddev"]) + fs + str(stats["variance"])
+            string += fs + str(stats["coeff_var"]) + fs + str(stats["sum"])
+            string += fs + str(stats["null_cells"]) + fs + str(stats["n"])
+            string += fs + str(int(stats["n"]))
```

Comment:
When rewriting portions of code like this, use f-strings where you can. No reason to be consistent with outdated code (it was never best practice AFAIK and now is probably flagged by linters).

The `string +=` you probably need to keep to minimize changes, but I think `strings.append(...)` and `"".join(strings)` is the solution there (theoretically faster, practically I did not test).